foolscap/connection_plugins.py

```diff
@@ -11,6 +12,8 @@
                                                         DNS_NAME_RESTR))
 NEW_STYLE_HINT_RE=re.compile(r"^tcp:(%s|%s):(\d+){1,5}$" % (DOTTED_QUAD_RESTR,
                                                             DNS_NAME_RESTR))
+SOCKS_HINT_RE=re.compile(r"^(tcp|tor):(%s|%s):(\d+){1,5}$" % (DOTTED_QUAD_RESTR,
```

Comment:
Both here and in `NEW_STYLE_HINT_RE` I would replace `(tcp|tor)` and `tcp` with `[^:]*` so that they can be registered for any hint type by the application (and we don't have to bind the SOCKS handler to Tor and TCP overriding).